HC-1220 Update "Check and update the firmware for WiFiNINA and WiFi101" (#211)
arduino/help-center-content: content/Hardware Support/Generic/Update-the-firmware-for-WiFiNINA-and-WiFi101.md

---
title: "Update the firmware for WiFiNINA and WiFi101"
id: 360013896579
---

Learn how to update the firmware for the WiFi101 and WiFiNINA communication libraries used on these boards:

> * Arduino MKR WiFi 1010
> * Arduino MKR Vidor 4000
> * Arduino Nano 33 IoT
> * Arduino UNO WiFi Rev2
> * Arduino Nano RP2040 Connect
> * Arduino MKR 1000 WiFi

Reviewer: ```suggestion
* Arduino MKR 1000 WiFi
* Arduino MKR Vidor 4000
* Arduino MKR WiFi 1010
* Arduino Nano 33 IoT
* Arduino Nano RP2040 Connect
* Arduino UNO WiFi Rev2
```
Sorted boards alphabetically.

> * [Arduino MKR WiFi 1010](https://store.arduino.cc/arduino-mkr-wifi-1010)
> * [Arduino MKR Vidor 4000](https://store.arduino.cc/arduino-mkr-vidor-4000)
> * [Arduino Nano 33 IoT](https://store.arduino.cc/arduino-nano-33-iot)
> * [Arduino UNO WiFi Rev2](https://store.arduino.cc/arduino-uno-wifi-rev2)
> * [Arduino Nano RP2040 Connect](https://store.arduino.cc/collections/boards/products/arduino-nano-rp2040-connect)
> * [Arduino MKR 1000 WiFi](https://docs.arduino.cc/hardware/mkr-1000-wifi)

Reviewer: ```suggestion
* Arduino MKR WiFi 1010
* Arduino MKR Vidor 4000
* Arduino Nano 33 IoT
* Arduino UNO WiFi Rev2
* Arduino Nano RP2040 Connect
* Arduino MKR 1000 WiFi
```

---
title: "Update the firmware for WiFiNINA and WiFi101"
id: 360013896579
---

> Learn how to update the firmware for WiFiNINA and WiFi101 using the Firmware Updater tool.

Reviewer: Since we also cover IoT Cloud as an option, in which case the term "Firmware Updater" doesn't really enter into it for the user, we can avoid the use of it here.

```suggestion
Learn how to update the firmware for the WiFi101 and WiFiNINA communication libraries.
```

I would consider including the supported boards here and dropping the "compatible boards" section:

```suggestion
Learn how to update the firmware for the WiFi101 and WiFiNINA communication libraries used on these boards:

* [Arduino MKR WiFi 1010](https://store.arduino.cc/arduino-mkr-wifi-1010)
* [Arduino MKR Vidor 4000](https://store.arduino.cc/arduino-mkr-vidor-4000)
* [Arduino Nano 33 IoT](https://store.arduino.cc/arduino-nano-33-iot)
* [Arduino UNO WiFi Rev2](https://store.arduino.cc/arduino-uno-wifi-rev2)
* [Arduino Nano RP2040 Connect](https://store.arduino.cc/collections/boards/products/arduino-nano-rp2040-connect)
* [Arduino MKR 1000 WiFi](https://docs.arduino.cc/hardware/mkr-1000-wifi)
```

In this article:

* [Compatible Boards](#compatible-boards)
* [Update the firmware in Arduino IDE 2](#update-the-firmware-in-arduino-ide-2)
* [Update the firmware in Arduino IoT Cloud](#update-the-firmware-in-arduino-iot-cloud)

---

<a id="compatible-boards"></a>

## Compatible Boards

> Boards with the **WiFiNINA** module:
> 
> * [Arduino MKR WiFi 1010](https://store.arduino.cc/arduino-mkr-wifi-1010)
> * [Arduino MKR Vidor 4000](https://store.arduino.cc/arduino-mkr-vidor-4000)
> * [Arduino Nano 33 IoT](https://store.arduino.cc/arduino-nano-33-iot)
> * [Arduino UNO WiFi Rev2](https://store.arduino.cc/arduino-uno-wifi-rev2)
> * [Arduino Nano RP2040 Connect](https://store.arduino.cc/collections/boards/products/arduino-nano-rp2040-connect)
> 
> Boards with the **WiFi101** module:
> 
> * [Arduino MKR 1000 WiFi](https://docs.arduino.cc/hardware/mkr-1000-wifi)

Reviewer: * It occurs to me that this article was confusing the library names (WiFi101 and WiFiNINA) with the names of the connectivity modules, which are different.
* Since we're dropping the instructions that involved opening an Example from the respective library, I don't think there's any need to emphesize this categorization—it can just be a single list of boards that can be used with the tool.

```suggestion
These boards uses the WiFi101 or WiFiNINA library:

* [Arduino MKR WiFi 1010](https://store.arduino.cc/arduino-mkr-wifi-1010)
* [Arduino MKR Vidor 4000](https://store.arduino.cc/arduino-mkr-vidor-4000)
* [Arduino Nano 33 IoT](https://store.arduino.cc/arduino-nano-33-iot)
* [Arduino UNO WiFi Rev2](https://store.arduino.cc/arduino-uno-wifi-rev2)
* [Arduino Nano RP2040 Connect](https://store.arduino.cc/collections/boards/products/arduino-nano-rp2040-connect)
* [Arduino MKR 1000 WiFi](https://docs.arduino.cc/hardware/mkr-1000-wifi)
```

Author: I agree. It also adds clarity to the article.

<a id="arduino-ide-2"></a>

## Update the firmware in Arduino IDE 2

To update the firmware in Arduino IDE 2 follow these steps:

1. Connect the board to your computer and open the Arduino IDE 2.

2. In the top menu bar, open **Tools > WiFi101 / WiFiNINA Firmware Updater**.

    ![Arduino IDE 2 with the WiFi101 / WiFiNINA Firmware Updater highlighetd under the Tools menu](img/firmware-updater-arduino-ide-2-menu-selection.png)

3. A new window will open. Select your board on the drop-down menu and click *CHECK UPDATES*

    ![WiFi101 / WiFiNINA Firmware Updater window displaying the board selection menu and the "check updates" button](img/firmware-updater-arduino-ide-2-check-updates.png)

4. Select the latest firmware version on the drop-down menu and click *INSTALL* (Note that the installation will overwrite any existing sketch on your board).

    ![WiFi101 / WiFiNINA Firmware Updater window displaying selected firmware version and the "install" button](img/firmware-updater-arduino-ide-2-select-board-install.png)

> 5. The "Firmware successfuly installed" dialogue will appear when the process is finished.

Reviewer: Is it really a dialogue? 

```suggestion
5. "Firmware successfuly installed" will be displayed when the process is finished.
```

    !["Firmware successfuly installed" message displaying on the WiFi101 / WiFiNINA Firmware Updater window](img/firmware-updater-arduino-ide-2-installation-successful.png)

> 

Reviewer: Since we are know calling the section "Update the firmware in Arduino IDE", could we keep the old instructions for IDE 1 here, under a **"Update the firmware in Arduino IDE 1"** heading?

Author: Good idea
